### PR TITLE
Added a generic method to create then run a pipeline based on the current method name

### PR DESCRIPTION
**Pull request by Vlatombe**
Resolves $NAME to the current name when used in the current pipeline script before running it. This makes the pipeline scripts more portable.

### CODE REVIEW

**Comment by Vlatombe** [comment]
Need to fix some remaining tests I didn't update w.r.t the changed groovy files

File: src/test/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/KubernetesPipelineTest.java [hunk 1]
```diff
@@ -342,7 +334,9 @@ public void runWithActiveDeadlineSeconds() throws Exception {
 
     @Test
     public void runInPodWithRetention() throws Exception {
+        logs.capture(1000);
         r.assertBuildStatusSuccess(r.waitForCompletion(b));
-        assertTrue(deletePods(cloud.connect(), getLabels(this, name), true));
+        logs.getMessages().stream().anyMatch(msg -> msg.contains("was not deleted due to retention policy Always"));
```
Comment (jglick) [inline src/test/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/KubernetesPipelineTest.java]: Related to this patch?

Comment (Vlatombe) [reply src/test/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/KubernetesPipelineTest.java]: argh, this got committed unexpectedly.

**Comment by jglick** [comment]
Merged this into my own PRs (#461, #490, #503, #522). Build failures seem to be a proxy issue: [INFRA-2164](https://issues.jenkins-ci.org/browse/INFRA-2164)

**Pull request merged**